### ISSUE TITLE
Windows only: Incorrect alignments and consensus sequences after clustering

### ISSUE THREAD

**Issue by torognes** [torognes/vsearch#275]
A user has reported that VSEARCH version 2.6.0 gave different consensus sequences after clustering on Windows 10 64-bit and Fedora 25 64-bit with this command:

`vsearch -cluster_smallmem  input.fasta -id 0.97 -consout consensus.fasta  -sizeout`

Some cluster abundances and consensus sequences were different between Windows and Linux.

**Comment by frederic-mahe** [comment]
could it be linked to multithreading?

**Comment by torognes** [comment]
I have received a dataset and have reproduced the error. It does not seem to be linked to multithreading. Results are equal on Linux and Mac, but different on windows.

**Comment by torognes** [comment]
It seems like the SIMD-based aligner (in align_simd.cc) produces a wrong alignment in some cases on Windows. This applies to searching, clustering, allpairs etc.


Running the following command

```
vsearch --allpairs_global mini.fasta --id 0 --alnout mini.aln
```

on the following minimalistic input file (mini.fasta)

```
>a
GCCGGGTCCCCCCCCCTCAGC
>b
GCCGGGTCCCCCCCACCCTCATC
```

gives different results. The correct result on Mac (and Linux):

```
vsearch --allpairs_global mini.fasta --id 0 --alnout mini.aln
vsearch v2.6.0_macos_x86_64, 16.0GB RAM, 8 cores

Query >a
 %Id   TLen  Target
 87%     23  b

 Query 21nt >a
Target 23nt >b

Qry  1 + GCCGGGTcccccc--cccTCAGC 21
         |||||||||||||  |||||| |
Tgt  1 + GCCGGGTCCCCCCCACCCTCATC 23

23 cols, 20 ids (87.0%), 2 gaps (8.7%)
```

The incorrect result on Windows:

```
vsearch --allpairs_global mini.fasta --id 0 --alnout mini.aln
vsearch v2.6.0_win_x86_64, 2.0GB RAM, 1 cores

Query >a
 %Id   TLen  Target
 87%     23  b

 Query 21nt >a
Target 23nt >b

Qry  1 + GCCGGGT-cccccc-cccTCAGC 21
         ||||||| |||||| |||||| |
Tgt  1 + GCCGGGTCCCCCCCACCCTCATC 23

23 cols, 20 ids (87.0%), 2 gaps (8.7%)
```

An extra gap is opened.

**Comment by torognes** [comment]
Fixed in latest commit. Was due to improper use suffices (UL instead of ULL) on 64-bit constants. On Windows a long (L) is 32-bits, while it is 64-bits on Macs and Linux. A long long (LL) is always 64 bits.

**Comment by torognes** [comment]
Fixed in VSEARCH 2.7.1 just released.
